src/modules/Login/action.js

```diff
@@ -135,11 +145,30 @@ export const initializeAccount = (account: EdgeAccount, touchIdInfo: Object) =>
     accountInitObject.merchantMode = syncFinal.merchantMode
     accountInitObject.customTokens = syncFinal.customTokens
     accountInitObject.passwordRecoveryRemindersShown = syncFinal.passwordRecoveryRemindersShown
-    Object.keys(syncFinal).forEach(currencyCode => {
-      if (typeof syncFinal[currencyCode].denomination === 'string') {
-        accountInitObject.denominationKeys.push({ currencyCode: currencyCode, denominationKey: syncFinal[currencyCode].denomination })
+
+    // begin process for repairing damaged settings data
+    let isOverwriteSyncedSettingsNeeded = false
+    const syncFinalKeys = Object.keys(syncFinal)
```

Comment:
@paullinator I think the issue with that approach is that when we call `setSyncedSettings` it saves the settings to disk, so I had to architect it to make sure we only call it once. Let me know if that makes sense.

Reply:
Correct. My suggestion above should only require you to call setSyncSettings once. the `syncFinal` object is the staging object to hold what you will eventually write to disk

Reply:
@paullinator I think I understand what you're saying and should be able to implement it without issue. So should we throw an error or fix the broken data, or both? I ask because if we throw an error but also fix it, then it may be difficult to debug what was broken

Reply:
@paullinator Unfortunately, creating a function that can handle all 3 cases (synced, local, and core) would need the following parameters:
`export const mergeSettings = (loadedSettings: object, defaults: object, types: object, account: object, type: string, accountInitObject: object = {}) => {`
Eventually we can phase out `accountInitObject` since that's just a matter of poor architecture, but whether we use `setSyncedSettings` or `setLocalSettings` will depend upon the `type` variable (ie `synced` or `local`), etc.

I can push a commit for not attempting to refactor the function, but it would take some time to properly create the function.

Reply:
Actually now that I think of it, I suppose I could pass the `setSyncedSettings` or `setLocalSettings` as a callback accessing `account` from the calling scope. Let me give that a try and see if it cleans it up enough.